stack.yaml

```diff
@@ -0,0 +1,22 @@
+resolver: lts-19.6
+
+ghc-options:
+  "$locals": >-
+    -fwrite-ide-info
+    -Weverything
+    -Wno-all-missed-specialisations
+    -Wno-deriving-typeable
+    -Wno-implicit-prelude
+    -Wno-incomplete-uni-patterns
+    -Wno-missed-specialisations
+    -Wno-missing-export-lists
+    -Wno-missing-exported-signatures
+    -Wno-missing-import-lists
+    -Wno-missing-local-signatures
+    -Wno-missing-monadfail-instances
+    -Wno-missing-safe-haskell-mode
+    -Wno-monomorphism-restriction
+    -Wno-partial-fields
+    -Wno-prepositive-qualified-module
+    -Wno-safe
+    -Wno-unsafe
```

Comment:
As I understand it, putting options into a `stack.yaml` (vs `package.yaml`) is preferred if:

- You're a multi-package project and you want to centralize these

  This doesn't apply here.

- You only want the options applied when this resolver is in use (i.e. not on older GHCs or `nightly`)

  I tend to work as strictly as is pragmatic, which means (absent major trade-offs) I want the `-Weverything` + opt-out approach even when I'm testing on older resolvers. I also want new warnings coming in on `nightly` to trip as early as possible.

  (This approach does mean a few `condition:` stanzas for options not known to older GHCs, but I've found it not too bad. I can find you an example if you want.)

Therefore, I've started putting these in `package.yaml`. Most of our existing OSS is putting it in `stack.yaml` just because I had no appreciation for the difference and copied it from somewhere.

WDYT about doing `package.yaml` in the template?

Reply:
This is from `freckle-app`.

Reply:
Gotcha. I think scientist might be a better example to start with:

https://github.com/freckle/scientist-hs/blob/main/package.yaml#L12-L25